prover/src/opcodes/comparison.rs

```diff
@@ -549,211 +557,1136 @@ impl TableFiller<ProverPackedField> for SleiuTable {
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use anyhow::Result;
-    use petravm_asm::isa::GenericISA;
-    use proptest::prelude::*;
-    use proptest::prop_oneof;
+/// SLT table.
+///
+/// This table handles the SLT instruction, which performs signed
+/// integer comparison (set if less than) between two 32-bit elements.
+pub struct SltTable {
+    id: TableId,
+    state_cols: StateColumns<SLT_OPCODE>,
+    dst_abs: Col<B32>,
+    src1_abs: Col<B32>,
+    src1_val: Col<B1, 32>,
+    src1_sign: Col<B1>,
+    src2_abs: Col<B32>,
+    src2_val: Col<B1, 32>,
+    src2_sign: Col<B1>,
+    dst_bit: Col<B1>,
+    subber: U32Sub,
+}
 
-    use crate::model::Trace;
-    use crate::prover::Prover;
-    use crate::test_utils::generate_trace;
+impl Table for SltTable {
+    type Event = SltEvent;
 
-    /// Creates an execution trace for a simple program that uses the SLEU or
-    /// SLTU instructions.
-    fn generate_unsigned_trace(src1_val: u32, src2_val: u32, op: &str) -> Result<Trace> {
-        let asm_code = format!(
-            "#[framesize(0x10)]\n\
-             _start: 
-                LDI.W @2, #{}\n\
-                LDI.W @3, #{}\n\
-                {} @4, @2, @3\n\
-                RET\n",
-            src1_val,
-            src2_val,
-            op.to_uppercase(),
+    fn name(&self) -> &'static str {
+        "SltTable"
+    }
+
+    // TODO: Consider swapping the order of src1 and src2 depending on the sign,
+    // or using a U32Add gadget.
```

Comment:
not sure I understand this comment, what would you like to achieve?

Reply:
Old comment